cobigen/cobigen-tsplugin/src/main/java/com/devonfw/cobigen/tsplugin/merger/TypeScriptMerger.java

```diff
@@ -63,107 +86,125 @@ public String getType() {
+                    importsAndExports.append(line);
+                    importsAndExports.append(LINE_SEP);
+                } else {
+                    body.append(line);
+                    body.append(LINE_SEP);
+                }
+            });
 
-            scriptEngines.put(scriptName, jsEngine);
+        } catch (IllegalStateException e) {
+
+            LOG.error("Closing connection on InputReader.", e);
+            request.terminateProcessConnection();
+        } catch (Exception e) {
+            connectionExc.setConnectExceptionMessage("Connection to server failed, attempt number " + 0 + ".");
+            connectionExc.setIOExceptionMessage("IO exception when merging");
+
+            connectionExc.handle(e);
         }
+        return runBeautifierExcludingImports(importsAndExports.toString(), body.toString());
+    }
 
-        Invocable jsInvocable = (Invocable) scriptEngines.get(scriptName);
-        try {
-            return (String) executable.exec(jsInvocable);
-        } catch (NoSuchMethodException e) {
-            throw new MergeException(base,
-                "Invalid API of " + scriptName + " script used. This is most properly a bug. Please report on Github.",
-                e);
-        } catch (ScriptException e) {
-            throw new MergeException(base, "Execution of the script " + scriptName + " raised an error.", e);
+    /**
+     * Sends a transfer object to the server, by using http connection
+     * @param dataTO
+     *            Data to transfer, it should be a serializable class
+     * @param conn
+     *            {@link HttpURLConnection} to the server, containing also the URL
+     * @param os
+     *            {@link OutputStream} that will be opened for sending data
+     * @param osw
+     *            {@link OutputStreamWriter} used for writing the data to be sent
+     * @throws IOException
+     *             when connection to the server failed
+     * @throws JsonGenerationException
+     *             When generating the JSON from the transfer object failed
+     * @throws JsonMappingException
+     *             When mapping the JSON from the transfer object failed
+     */
+    private void sendRequest(Object dataTO, HttpURLConnection conn, OutputStream os, OutputStreamWriter osw)
+        throws IOException, JsonGenerationException, JsonMappingException {
+        ObjectWriter objWriter;
+        objWriter = new ObjectMapper().writer().withDefaultPrettyPrinter();
+        String jsonMergerTO = objWriter.writeValueAsString(dataTO);
+
+        // We need to escape new lines because otherwise our JSON gets corrupted
+        jsonMergerTO = jsonMergerTO.replace("\\n", "\\\\n");
+
+        osw.write(jsonMergerTO);
+        osw.flush();
+        os.close();
+        conn.connect();
+
+        if (conn.getResponseCode() != HttpURLConnection.HTTP_CREATED) {
+            throw new RuntimeException("Failed : HTTP error code : " + conn.getResponseCode());
```

Comment:
Please make use of the CobiGen Exception (CobiGenRuntimeException), this would differentiate exceptions accidentially occurred and the ones we fired explicitly

Reply:
Maybe this is also an exception, we should serve by the cobigen API because if there are connection problems, we may want to implement some retry mechanism in future in cobigen-core. So the exception should be easy to be differentiated.

Reply:
I have created this HTTP exception handling [here](https://github.com/devonfw/tools-cobigen/pull/840/files#diff-45a2daf5ecab0802bb8ac0f498aa8a99). Can you check if this is what you meant?

It returns true when the HTTP request was 200 or 201, and false otherwise.

Reply:
No, I just meant, that on any download error, we might want to serve a secific type of exception, which we can handle to make retries. Therefore CobiGenRuntimeException might be too generic to not retry NPE for example.